### PR TITLE
Add radosgw_frontend_ssl_certificate parameter

### PR DESCRIPTION
**Pull request by gfidente**
This is necessary when configuring RGW/civetweb with SSL because
in addition to passing a specific frontend option, we need to
append the 's' character to the binding port.

Signed-off-by: Giulio Fidente <gfidente@redhat.com>

### CODE REVIEW

File: roles/ceph-config/templates/ceph.conf.j2 [hunk 1]
```diff
@@ -121,7 +121,7 @@ osd memory target = {{ _osd_memory_target | default(osd_memory_target) }}
 host = {{ _rgw_hostname }}
 keyring = /var/lib/ceph/radosgw/{{ cluster }}-rgw.{{ _rgw_hostname + '.' + instance['instance_name'] }}/keyring
 log file = /var/log/ceph/{{ cluster }}-rgw-{{ hostvars[host]['ansible_hostname'] + '.' + instance['instance_name'] }}.log
-rgw frontends = {{ radosgw_frontend_type }} {{ 'port' if radosgw_frontend_type == 'civetweb' else 'endpoint' }}={{ instance['radosgw_address'] }}:{{ instance['radosgw_frontend_port'] }} {{ radosgw_frontend_options }}
+rgw frontends = {{ radosgw_frontend_type }} {{ 'port' if radosgw_frontend_type == 'civetweb' else 'endpoint' }}={{ instance['radosgw_address'] }}:{{ instance['radosgw_frontend_port'] }}{{ 's ssl_certificate='+radosgw_civetweb_ssl_certificate if radosgw_civetweb_ssl_certificate }} {{ radosgw_frontend_options }}
```
Comment (dsavineau) [inline roles/ceph-config/templates/ceph.conf.j2]: This doesn't work becaise you need a else statement

```console
AnsibleUndefinedVariable: the inline if-expression on line 1 evaluated to false and no else section was defined.
```
Alternatively you could use something like :

```yaml
{% if radosgw_civetweb_ssl_certificate %}s ssl_certificate={{ radosgw_civetweb_ssl_certificate }}{% endif %}
```

But the real issue there is that this code will be also true for beast frontend if `radosgw_civetweb_ssl_certificate` is set but the ssl parameters are different.

Comment (gfidente) [reply roles/ceph-config/templates/ceph.conf.j2]: maybe we should use a bigger condition and set the parameters based on _ssl_certificate and _frontend_type ?

Comment (gfidente) [reply roles/ceph-config/templates/ceph.conf.j2]: updated with fixes as per comments and to deal better with both civetweb and beast

**Comment by dsavineau** [comment]
The jinja template still fails.
Also for beast you need the ssl key in a different file so there's another option for that.

http://docs.ceph.com/docs/master/radosgw/frontends/#options

**Comment by dsavineau** [comment]
Nevermind, I didn't read the documentation correctly. So it's fine with only one combined file for both civetweb and beast

**Comment by ktdreyer** [comment]
Do we need to backport this to `stable-4.0` for OSP, @gfidente ?

**Comment by dsavineau** [comment]
@ktdreyer According to https://bugzilla.redhat.com/show_bug.cgi?id=1722071 yes

**Comment by dsavineau** [comment]
@gfidente the code looks good but there's some part missing

- The ssl file needs to be available within the radosgw container. So we probably need to add a bind mount around there [1]. Note that we should change `ansible_distribution` by `ansible_os_family`.
- The radosgw handler needs to be updated to reflect the protocol [2].

[1] https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-rgw/templates/ceph-radosgw.service.j2#L31-L33
[2] https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-handler/templates/restart_rgw_daemon.sh.j2#L60

**Comment by gfidente** [comment]
@dsavineau thanks for the review, will do

File: roles/ceph-rgw/templates/ceph-radosgw.service.j2 [hunk 2]
```diff
@@ -31,6 +31,9 @@ ExecStart=/usr/bin/{{ container_binary }} run --rm --net=host \
   {% if ansible_distribution == 'RedHat' -%}
   -v /etc/pki/ca-trust/extracted:/etc/pki/ca-trust/extracted:z \
   {% endif -%}
+  {% if radosgw_frontend_ssl_certificate != '' -%}
+  -v {{ radosgw_frontend_ssl_certificate }}:ro \
```
[the remark this answers is not in the document]
Comment (gfidente) [reply roles/ceph-rgw/templates/ceph-radosgw.service.j2]: heh missed the paste, will fix

File: roles/ceph-handler/templates/restart_rgw_daemon.sh.j2 [hunk 3]
```diff
@@ -5,6 +5,12 @@ DELAY="{{ handler_health_rgw_check_delay }}"
 HOST_NAME="{{ ansible_hostname }}"
 RGW_NUMS={{ radosgw_num_instances }}
 RGW_BASE_PORT={{ radosgw_frontend_port }}
+RGW_FRONTEND_SSL_CERT={{ radosgw_frontend_ssl_certificate }}
+if [ ! -z "$RGW_FRONTEND_SSL_CERT" ]; then
```
[the remark this answers is not in the document]
Comment (gfidente) [reply roles/ceph-handler/templates/restart_rgw_daemon.sh.j2]: ack will do

File: roles/ceph-rgw/templates/ceph-radosgw.service.j2 [hunk 4]
```diff
@@ -31,6 +31,9 @@ ExecStart=/usr/bin/{{ container_binary }} run --rm --net=host \
   {% if ansible_distribution == 'RedHat' -%}
   -v /etc/pki/ca-trust/extracted:/etc/pki/ca-trust/extracted:z \
   {% endif -%}
+  {% if radosgw_frontend_ssl_certificate != '' -%}
```
[the remark this answers is not in the document]
Comment (guits) [reply roles/ceph-rgw/templates/ceph-radosgw.service.j2]: @gfidente

**Comment by dsavineau** [comment]
jenkins test pipeline

**Comment by dsavineau** [comment]
mergify failed to merge, doing it manually

**Pull request merged**